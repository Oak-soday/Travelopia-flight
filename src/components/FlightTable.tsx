import React, { useState, useEffect } from 'react';
import FlightRow from './FlightRow';
import { fetchFlights, Flight } from '../api/flights';
import './css/flighttable.css';
import { Outlet } from 'react-router-dom';
import loadingIcon from '../assets/flight.png';

const FlightTable: React.FC = () => {
    const [flights, setFlights] = useState<Flight[]>([]);
    useEffect(() => {
        const fetchFlightData = async () => {
            try {
                const data = await fetchFlights();
                setFlights(data);
            } catch (error) {

            }
        };
        fetchFlightData();
        const interval = setInterval(fetchFlightData, 1500);
        return () => clearInterval(interval);
    }, []);

    return (
        <div data-testid="flight-table" className='flight-container'>
            {flights.length != 0 ?
                <table className="flight-table">
                    <thead>
                        <tr>
                            <th>Flight Number</th>
                            <th>Airline</th>
                            <th>Origin</th>
                            <th>Destination</th>
                            <th>Departure Time</th>
                            <th>Status</th>
                        </tr>
                    </thead>
                    <tbody>
                        {flights.map((flight) => (
                            <FlightRow data-testid="flight-list" key={flight.id} flight={flight} />
                        ))}
                    </tbody>
                </table>
                : <img className="loading-icon" src={loadingIcon}></img>}
            <Outlet></Outlet>
        </div>
    );
};

export default FlightTable;
