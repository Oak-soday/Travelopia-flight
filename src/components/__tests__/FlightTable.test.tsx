
import renderer from 'react-test-renderer';
import { test, expect } from 'vitest';
import FlightTable from '../FlightTable';
import { render, screen } from '@testing-library/react';

test('mount component', async () => {
    expect(FlightTable).toBeTruthy();
});

test('screenshot testing', () => {
    const tree = renderer
        .create(
            <FlightTable></FlightTable>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
test("should render flight Table", () => {
    render(<FlightTable></FlightTable>);
    const FlightElement = screen.getByTestId('flight-table');
    expect(FlightElement).toBeDefined();
});


