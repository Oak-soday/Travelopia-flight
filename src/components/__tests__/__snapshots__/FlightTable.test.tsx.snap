// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`screenshot testing 1`] = `
<div
  data-testid="flight-table"
>
  <img
    className="loading-icon"
    src="/src/assets/flight.png"
  />
</div>
`;
